### ISSUE TITLE
Improve handling of PBXLegacyTarget

### ISSUE THREAD

**Issue by MasterCarl** [CocoaPods/CocoaPods#3616]
### Report

* What did you do?
Update from a pre-release version of BrightFutures
* What did you expect to happen?
All pods up to date
* What happened instead?
See the error message below

### Stack

```
   CocoaPods : 0.37.2
        Ruby : ruby 2.0.0p481 (2014-05-08 revision 45883) [universal.x86_64-darwin14]
    RubyGems : 2.0.14
        Host : Mac OS X 10.10.3 (14D136)
       Xcode : 6.3.2 (6D2105)
         Git : git version 2.1.2
Ruby lib dir : /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib
Repositories : master - https://github.com/CocoaPods/Specs.git @ 27816f7d0ed2e6607c403487c39f52845cf1f684
```

### Plugins

```
cocoapods-plugins : 0.4.2
cocoapods-trunk   : 0.6.1
cocoapods-try     : 0.4.5
```

### Podfile

```ruby
# Uncomment this line to define a global platform for your project
source 'https://github.com/CocoaPods/Specs.git'
use_frameworks!
platform :ios, '8.0'

target 'CR' do
	pod 'CocoaLumberjack', '2.0.0-rc'
	pod 'BrightFutures', '1.0.1'
	pod 'JGProgressHUD', '~> 1.2'
	pod 'MBBarMenu', '~> 0.1.1'
	pod 'MBSegueCollection', '~> 0.2.1'
end

target 'CRTests' do
	pod 'CocoaLumberjack', '2.0.0-rc'
end

target 'CRProvider' do
	pod 'CocoaLumberjack', '2.0.0-rc'
	pod 'BrightFutures', '1.0.1'
	pod 'JGProgressHUD', '~> 1.2'
end

target 'CRProviderFileProvider' do
	pod 'CocoaLumberjack', '2.0.0-rc'
	pod 'BrightFutures', '1.0.1'
	pod 'JGProgressHUD', '~> 1.2'
end

target 'StaticLibs' do

end
```

### Error

```
NoMethodError - undefined method `symbol_type' for #<Xcodeproj::Project::Object::PBXLegacyTarget:0x007fbddbe18458>
/Library/Ruby/Gems/2.0.0/gems/cocoapods-0.37.2/lib/cocoapods/installer.rb:554:in `map'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-0.37.2/lib/cocoapods/installer.rb:554:in `block in set_target_dependencies'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-0.37.2/lib/cocoapods/installer.rb:553:in `each'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-0.37.2/lib/cocoapods/installer.rb:553:in `set_target_dependencies'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-0.37.2/lib/cocoapods/installer.rb:136:in `block in generate_pods_project'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-0.37.2/lib/cocoapods/user_interface.rb:59:in `section'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-0.37.2/lib/cocoapods/installer.rb:132:in `generate_pods_project'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-0.37.2/lib/cocoapods/installer.rb:99:in `install!'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-0.37.2/lib/cocoapods/command/project.rb:71:in `run_install_with_update'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-0.37.2/lib/cocoapods/command/project.rb:155:in `run'
/Library/Ruby/Gems/2.0.0/gems/claide-0.8.1/lib/claide/command.rb:312:in `run'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-0.37.2/lib/cocoapods/command.rb:46:in `run'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-0.37.2/bin/pod:44:in `<top (required)>'
/usr/bin/pod:23:in `load'
/usr/bin/pod:23:in `<main>'
```

**Comment by neonichu** [comment]
Is one of the targets you're integrating an external build target by any chance?

**Comment by MasterCarl** [comment]
I'm not really sure. However, removing the empty target resolved the issue.

**Comment by neonichu** [comment]
Does the icon of the "StaticLibs" target look like this?

![screen shot 2015-05-28 at 20 38 03](https://cloud.githubusercontent.com/assets/118992/7867984/885c0fe2-0579-11e5-9316-b0e18f07656b.png)

**Comment by MasterCarl** [comment]
Turns out it does.

**Comment by neonichu** [comment]
Thanks for checking, I guess CP should provide an error or skip those build targets, as integrating them doesn't make any sense.

I'd say it should skip it, @kylef @orta any opinions?

**Comment by segiddins** [comment]
Skip + warn?

**Comment by orta** [comment]
Skip + Warn I think.

**Comment by kylef** [comment]
Agreed with skipping and warning :ok_hand:

**Comment by segiddins** [comment]
This might actually be fixed on 0.38.0.beta.1, @MasterCarl, if you want to give that a try?

**Comment by segiddins** [comment]
Closing due to lack of a response.
